android-core/src/main/java/com/mparticle/modernization/core/MParticle.kt

```diff
@@ -0,0 +1,31 @@
+package com.mparticle.modernization.core
+
+import com.mparticle.MParticleOptions
+import com.mparticle.modernization.data.MParticleDataRepository
+import com.mparticle.modernization.eventlogging.MParticleEventLogging
+import com.mparticle.modernization.identity.MParticleIdentity
+import com.mparticle.modernization.kit.MParticleKitManager
+import com.mparticle.modernization.datahandler.MParticleDataHandler
+
+internal class MParticle private constructor(private val options: MParticleOptions) {
```

Comment:
What exactly does `private val` do in this constructor parameter? Is it syntax sugar for creating a private instance property called `options` and assigning it this value?

Reply:
This enables to use the dependency injection pattern, making the property accesible to be used in the class but restrict visibility from consumers. If options would have been public, MParticle.getInstance().options would have been valid.. Moreover by default properties have public setters/getters so, someone from the outside could eventually temper with the properties within MParticleOptions